vite_ruby/lib/vite_ruby/builder.rb

```diff
@@ -14,7 +14,10 @@ def build(*args)
     last_build = last_build_metadata(ssr: args.include?('--ssr'))
 
     if args.delete('--force') || last_build.stale?
-      build_with_vite(*args).tap { |success| record_build_metadata(success, last_build) }
+      build_with_vite(*args).yield_self do |success, err_msg|
+        record_build_metadata(success, last_build, err_msg)
+        success
+      end
```

Comment:
It's probably a good time to refactor `capture` to return a boolean instead of the status code.

```ruby
stdout, stderr, success = build_with_vite(*args)
log_build_result(stdout, stderr, success)
record_build_metadata(last_build, stderr, success)
success
```